src/main/java/com/dougnoel/sentinel/files/CsvFile.java

```diff
@@ -0,0 +1,334 @@
+        super(pathToFile);
+        csvFormat = CSVFormat.DEFAULT;
+        numHeaderRows = 1;
+    }
+
+    /**
+     * Create a CSV file with the given file format from the CSV file at the specified path.
+     * @param pathToFile Path path to the CSV file.
+     * @param format CSVFormat the format of the CSV file. See CSVFormat documentation for more info.
+     * @throws FileNotFoundException in the case that a file does not exist at that path.
+     */
+    public CsvFile(Path pathToFile, CSVFormat format) throws FileNotFoundException {
+        super(pathToFile);
+        csvFormat = format;
+        numHeaderRows = 1;
+    }
+
+    /**
+     * Create a CSV file configured with the given number of header rows from the CSV file at the most recently downloaded path.
+     * @param numberOfHeaderRows int number of header rows in the CSV file.
+     * @throws FileNotFoundException in the case that a file does not exist at that path.
+     */
+    public CsvFile(int numberOfHeaderRows) throws FileNotFoundException {
+        super();
+        csvFormat = CSVFormat.DEFAULT;
+        numHeaderRows = numberOfHeaderRows;
+    }
+
+    /**
+     * Create a CSV file configured with the given number of header rows from the CSV file at the given path.
+     * @param pathToFile Path path to the CSV file.
+     * @param numberOfHeaderRows int number of header rows in the CSV file.
+     * @throws FileNotFoundException in the case that a file does not exist at that path.
+     */
+    public CsvFile(Path pathToFile, int numberOfHeaderRows) throws FileNotFoundException {
+        super(pathToFile);
+        csvFormat = CSVFormat.DEFAULT;
+        numHeaderRows = numberOfHeaderRows;
+    }
+
+    /**
+     * Create a CSV file configured with the given number of header rows and CSVFormat from the CSV file at the given path.
+     * @param pathToFile Path path to the CSV file.
+     * @param format CSVFormat the format of the CSV file. See CSVFormat documentation for more info.
+     * @param numberOfHeaderRows int number of header rows in the CSV file.
+     * @throws FileNotFoundException in the case that a file does not exist at that path.
+     */
+    public CsvFile(Path pathToFile, CSVFormat format, int numberOfHeaderRows) throws FileNotFoundException {
+        super(pathToFile);
+        csvFormat = format;
+        numHeaderRows = numberOfHeaderRows;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        if (!super.equals(o)) return false;
+        CsvFile csvFile = (CsvFile) o;
+        return numHeaderRows == csvFile.numHeaderRows && Objects.equals(csvFormat, csvFile.csvFormat);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), csvFormat, numHeaderRows);
+    }
+
+    private CSVParser getParser() throws IOException {
```

Comment:
This is mainly due to the nature of the CSV library I'm using. The parser is essentially an iterator over the file, which cannot go backwards. That means if we kept the same parser object between method calls it could miss parts of the file. If method call 1 was to read the 3rd row, and method call 2 was to get all contents of the file, method call 2 would not return the first 3 rows. We have to make a new parser for each method call. This is also advantageous because it means every time we read from the file, we have the most up-to-date contents, and we don't run into access conflicts. when we're between reads, we are not technically accessing the file. So other processes are free to change it.